### PR TITLE
Make customizable base container image available on GitHub Container Registry

### PR DESCRIPTION
**Pull request by andrewnicolalde**
This pull request makes a customizable base container image for Starbase available to users via [GitHub Container Registry](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry).

Making these images available in a container registry is useful because, among other reasons, Starbase users will always have a reference environment in which to run Starbase, i.e. with the same OS version and OS package versions.

This differs from the [current setup](https://github.com/JupiterOne/starbase/blob/main/Dockerfile) where each user simply clones the Starbase repo and builds a container image, installing whatever OS package versions happen to be current at the time they build the image, which obviously could vary between users of the same release of Starbase.

Since user configuration will obviously differ between users, this image does not include a `config.yaml`, since it wouldn't be meaningful in this context. Instead, [a section has been added to the README](https://github.com/JupiterOne/starbase/commit/ff76fa573793834a64597a88336d8737b7e68c15) explaining how users should supply configuration to a container based on this image.

Similar PR made for Lyft's Cartography project: https://github.com/lyft/cartography/pull/851

### CODE REVIEW

**Comment by andrewnicolalde** [comment]
These checks should now pass after the `gitleaks` action was removed in https://github.com/JupiterOne/starbase/pull/72, but I can't seem to figure out how to actually re-run these checks in this PR :/

**Review by adam-in-ict — APPROVED** [review]

**Comment by andrewnicolalde** [comment]
Thanks for the approval! Looks like I don't have permissions to merge this, so could you merge it for me @adam-in-ict ?

File: .github/workflows/build.yml
Comment (austinkelleher) [inline .github/workflows/build.yml:31]: I wonder if it would be helpful to actually have a CI test that validates that Starbase can run from inside of the CI's built base image. This would help ensure that we don't introduce regressions into the Docker image.

**Review by austinkelleher — APPROVED** [review]
This looks good to me! Thank you so much for contributing this @andrewnicolalde!

**Comment by andrewnicolalde** [comment]
Looks like while the image did get built and pushed to GHCR, JupiterOne has configured starbase's packages to be private by default. [Looks like you can change the starbase package visibility this way, which should make the starbase container image public.](https://docs.github.com/en/packages/learn-github-packages/configuring-a-packages-access-control-and-visibility#configuring-visibility-of-container-images-for-an-organization)

**Comment by austinkelleher** [comment]
@andrewnicolalde Thanks for the follow-up. We are currently waiting for our security team to approve this change. We'll update when the image is public.

File: .github/workflows/build.yml
Comment (andrewnicolalde) [reply .github/workflows/build.yml:31]: Not a bad idea at all. How extensively do you reckon the test should verify that? I.e. actually run some starbase graph integrations or just verify that `yarn starbase setup` etc. works?

**Comment by austinkelleher** [comment]
@andrewnicolalde The image is now public. Let us know how it goes! Thanks again.

**Pull request merged**
